client/app/types/course/assessment/questions.ts

```diff
@@ -0,0 +1,107 @@
+export interface QuestionData {
+  id: number;
+  number: number;
+  defaultTitle: string;
+  title: string | null;
+  unautogradable: boolean;
+  type: string;
+
+  description?: string;
+  editUrl?: string;
+  deleteUrl?: string;
+  duplicationUrls?: {
+    tab: string;
+    destinations: {
+      title: string;
+      duplicationUrl: string;
+    }[];
+  }[];
+}
+
+interface OptionListData {
+  id: number | string;
+  option: string;
+  correct: boolean;
+}
+
+export interface OptionData extends OptionListData {
+  explanation: string;
+  weight: number;
+  ignoreRandomization: boolean;
+  toBeDeleted?: boolean;
+  draft?: boolean;
+}
+
+export interface McqMrqOptions {
+  type: string;
+  mcqMrqType: 'mcq' | 'mrq';
+  convertUrl: string;
+  hasAnswers: boolean;
+  unsubmitAndConvertUrl?: string;
+}
+
+export type McqMrqListData = QuestionData &
+  McqMrqOptions & {
+    options: OptionListData[];
+  };
+
+type OptionalIfNew<T extends 'new' | 'edit'> = T extends 'new'
+  ? undefined
+  : never;
+
+export interface McqMrqData<T extends 'new' | 'edit' = 'edit'> {
+  gradingScheme: 'any_correct' | 'all_correct';
+  options: OptionData[] | OptionalIfNew<T>;
+  question:
+    | {
+        title: string | null;
+        description: string | null;
+        staffOnlyComments: string | null;
+        maximumGrade: string;
+        skipGrading: boolean;
+        skillIds: number[];
+        randomizeOptions?: boolean;
+      }
+    | OptionalIfNew<T>;
+}
+
+export interface McqMrqFormData<T extends 'new' | 'edit' = 'edit'>
+  extends McqMrqOptions,
```

Comment:
No, because `McqMrqData` is used as the form's fields' data. It should only contain data that the form submits. Then again, I am not trying to mirror the back-end templates to the front-end types.